src/flask/blueprints.py

```diff
@@ -235,20 +236,40 @@ def register(self, app, options, first_registration=False):
         # Merge blueprint data into parent.
         if first_registration:
 
-            def extend(bp_dict, parent_dict):
+            def extend(bp_dict, parent_dict, ensure_sync=False):
                 for key, values in bp_dict.items():
                     key = self.name if key is None else f"{self.name}.{key}"
+                    if ensure_sync:
+                        values = [app.ensure_sync(func) for func in values]
                     parent_dict[key].extend(values)
 
             def update(bp_dict, parent_dict):
                 for key, value in bp_dict.items():
                     key = self.name if key is None else f"{self.name}.{key}"
+                    value = defaultdict(
+                        dict,
+                        {
+                            code: {
+                                exc_class: app.ensure_sync(func)
```

Comment:
I know this is only used for the error handler spec, but this change makes the `update` method not generic, so it needs a different name.